features/native_api.feature

```diff
@@ -0,0 +1,191 @@
+        And the event "user.email" equals "j@example.com"
+        And the event "unhandled" is true
+
+    Scenario: Notifying in C
+        When I run "CXXNotifyScenario"
+        And I wait a bit
+        Then I should receive a request
+        And the request payload contains a completed native report
+        And the event "severity" equals "error"
+        And the event "context" equals "MainActivity"
+        And the exception "errorClass" equals "Vitamin C deficiency"
+        And the exception "message" equals "9 out of 10 adults do not get their 5-a-day"
+        And the event "unhandled" is false
+
+    Scenario: Changing intents followed by notifying in C
+        When I run "CXXAutoContextScenario"
+        And I wait a bit
+        Then I should receive a request
+        And the request payload contains a completed native report
+        And the event "severity" equals "info"
+        And the event "context" equals "SecondActivity"
+        And the exception "errorClass" equals "Hello hello"
+        And the exception "message" equals "This is a new world"
+        And the event "unhandled" is false
+
+    Scenario: Update context in Java followed by crashing in C
+        When I run "CXXUpdateContextCrashScenario"
+        And I wait a bit
+        And I configure the app to run in the "non-crashy" state
+        And I relaunch the app
+        Then I should receive a request
+        And the request payload contains a completed native report
+        And the event "severity" equals "error"
+        And the event "context" equals "Everest"
+        And the exception "errorClass" equals "SIGILL"
+        And the event "unhandled" is true
+
+    Scenario: Leaving a breadcrumb followed by notifying in C
+        When I run "CXXBreadcrumbScenario"
+        And I wait a bit
+        Then I should receive a request
+        And the request payload contains a completed native report
+        And the event "severity" equals "info"
+        And the exception "errorClass" equals "Bean temperature loss"
+        And the exception "message" equals "100% more microwave required"
+        And the event has a "log" breadcrumb named "Cold beans detected"
+        And the event "unhandled" is false
+
+    Scenario: Leaving a breadcrumb followed by a C crash
+        When I run "CXXNativeBreadcrumbNativeCrashScenario"
+        And I configure the app to run in the "non-crashy" state
+        And I relaunch the app
+        Then I should receive a request
+        And the request payload contains a completed native report
+        And the event has a "request" breadcrumb named "Substandard nacho error"
+        And the exception "errorClass" equals "SIGILL"
+        And the event "severity" equals "error"
+        And the event "unhandled" is true
+
+    Scenario: Starting a session, notifying, followed by a C crash
+        When I run "CXXSessionInfoCrashScenario"
+        And I wait a bit
+        And I wait a bit
```

Comment:
If this is intentional I think we should just alter this step so that it optionally allows specifying how long to wait.

Reply:
It’s intentional. This one needs to send four requests. The relative wait time works well because it needs longer on lower powered devices/emulators.